aequilibrae/project/network/osm/osm_builder.py

```diff
@@ -1,263 +1,239 @@
+                # when doing the join with the link fields below
+                intersecs = np.where(node_indices > 1)[0]
+                for i, j in zip(intersecs[:-1], intersecs[1:]):
+                    geometries.append([idx, self._build_geometry(link.nodes[i : j + 1])])
+
+        # Builds the link Geo dataframe
+        self.links_df.drop(columns=["nodes"], inplace=True)
+        # We build a dataframe with the geometries created above
+        # and join with the database
+        geo_df = pd.DataFrame(geometries, columns=["link_id", "geometry"]).set_index("link_id")
+        self.links_df = self.links_df.join(geo_df, how="inner")
+
+        self.links_df.loc[:, "link_id"] = np.arange(self.links_df.shape[0]) + 1
+
+        self.node_df = self.node_df.reset_index()
+
+        # Saves the data to disk in case of issues loading it to the database
+        osm_data_path = Path(self.project.project_base_path) / "osm_data"
+        osm_data_path.mkdir(exist_ok=True)
+        self.links_df.to_parquet(osm_data_path / "links.parquet")
+        self.node_df.to_parquet(osm_data_path / "nodes.parquet")
+
+        self.logger.info("Adding nodes to file")
+        self.__emit_all(["text", "Adding nodes to file"])
+
+        # Removing the triggers before adding all nodes makes things a LOT faster
+        remove_triggers(conn, self.logger, "network")
+
+        cols = ["node_id", "osm_id", "is_centroid", "modes", "link_types", "lon", "lat"]
+        insert_qry = f"INSERT INTO nodes ({','.join(cols[:-2])}, geometry) VALUES(?,?,?,?,?, MakePoint(?,?, 4326))"
+        conn.executemany(insert_qry, self.node_df[cols].to_records(index=False))
 
-            if len(vars["modes"]) > 0:
-                for i in range(segments):
-                    attributes = self.__build_link_data(vars, intersections, i, linknodes, node_ids, fields)
-                    if attributes is None:
-                        continue
-                    all_attrs.append(attributes)
-                    vars["link_id"] += 1
-
-            self.__emit_all(["text", f"{counter:,} of {L:,} super links added"])
-            self.links[osm_id] = []
-        sql = self.insert_qry.format(table, field_names, ",".join(["?"] * (len(all_attrs[0]) - 1)))
-        self.logger.info("Adding network links")
-        self.__emit_all(["text", "Adding network links"])
-        try:
-            conn.executemany(sql, all_attrs)
-        except Exception as e:
-            self.logger.error("error when inserting link {}. Error {}".format(all_attrs[0], e.args))
-            self.logger.error(sql)
-            raise e
-
-    def __worksetup(self):
-        self.__link_types = self.project.network.link_types
-        lts = self.__link_types.all_types()
-        for lt_id, lt in lts.items():
-            self.__model_link_types.append(lt.link_type)
-            self.__model_link_type_ids.append(lt_id)
+        del self.node_df
+        gc.collect()
```

Comment:
Are these calls to `gc.collect()` shown to reduce memory usage? If we've derived something from this df then ensuring Python gets that memory back might be a little trickery that just `del` and `gc.collect()`

I don't believe this would be hurting us in any manner, more like that it might just be unneeded.